### DISCUSSION TITLE
serialization problem, from_json need construct new object

### DISCUSSION THREAD

**Discussion by jiapengwen** [nlohmann/json#3168] [Q&A] [answered]
show you my code,:
```c++
using json = nlohmann::json;

namespace ns {
    void to_json(json& j, const person& p) {
    }
  // here will create new object, but i just want to init my object(outside create) whit json value
    void from_json(const json& j, person& p) {
    }
} // namespace ns


// in othre file
person my_person;
//here init my_person with json (j) value
from_json(j, my_person);
```
thank you your help

**Comment by nlohmann** [comment]
I do not fully understand your problem - can you share more code? One thing though: you don't want to call `from_json` yourself. It is called automatically when you write code like:

```cpp
json j; // initialize with whatever value you want
person my_person = j; // this calls from_json(j, my_person) under the hood
```

**Reply by jiapengwen** [reply]
my_person = person();
json j = my_person;

// here create new object p, I want use json init already create object;
person p = j.get<person>();

// like this
person already_person;
//user already create object
from_json(j, already_person);

**Comment by jiapengwen** [comment, marked as answer]
I just want deserialization json to my already create object, but from_fson will create new object, this isn't my purpose.
```c++
my_person = person();
json j = my_person;

// here create new object p, I want use json init already create object;
person p = j.get<person>();

// like this
person already_person;
//user already create object
from_json(j, already_person);
```

In other word , I need from_json convert has no default constructor,

**Reply by nlohmann** [reply]
If your type is not default-constructible, then this may help: https://json.nlohmann.me/features/arbitrary_types/#how-can-i-use-get-for-non-default-constructiblenon-copyable-types

Otherwise, I still do not understand the issue.

**Reply by jiapengwen** [reply]
when I use from_json, I do not want to **create new object**.
```c++
namespace ns {
struct person {
  std::string name;
  std::string address;
  int age;
  deque<string> d;
  deque<tuple<int, double>> dd;
  NLOHMANN_DEFINE_TYPE_INTRUSIVE(person, name, address, age, d, dd);
};
}  // namespace ns

int main(){
  ns::person p{"Ned Flanders", "744 Evergreen Terrace", 60, {"a", "b", "c"}, {{1, 1.1}, {2, 2.2}}};
  json j = p;

  // the import,  I don't want  use t
  auto t = j.get<ns::person>();

  // this is my desire, use a already create object to load json value, In the other word,  load checkpoint,
  person xx;
  j.get<ns::person>(xx);
  xx == p;
}

```
I try my best describe my question,  thanks

**Reply by gregmarr** [reply]
Sounds like you want 

    person xx;
    j.get_to(xx);

Though that really isn't any different than

    auto xx = j.get<ns::person>();

unless `person` is expensive to create and you already have one, or it is part of some other already-created structure.

**Reply by jiapengwen** [reply]
yes, this  is what i want, In my case, it's useful

**Reply by jiapengwen** [reply]
In other case, one class instance has many data member, but only a few need serializaion, so I don't want create new instance
